File: 266-palindrome-permutation/266-palindrome-permutation.cpp
class Solution {
public:
    bool canPermutePalindrome(string s) {
        int chr[26] = {0};
        
        for(char ch:s)
            chr[ch-'a']++;
        
        int count=0;
        for(int i=0;i<26;i++)
        {
            if(chr[i]%2!=0)
                count++;
        }
        if(count<=1)
            return true;
        else
            return false;
    }
};

File: 266-palindrome-permutation/266-palindrome-permutation.java
class Solution {
    public boolean canPermutePalindrome(String s) {
        HashMap<Character, Integer> chr =  new HashMap<>();
        char[] str = s.toCharArray();
        for(char ch: str)
        {
            //chr.put(Character.valueOf(ch),chr.getOrDefault(Character.valueOf(ch),0)+1);
            chr.put(ch,chr.getOrDefault(ch,0)+1);
        }
        int count = 0;
        for(char key: chr.keySet())
        {
            count+=chr.get(key)%2;
        }
        return count<=1;
    }
}